server/src/test/java/org/elasticsearch/discovery/AbstractDisruptionTestCase.java

```diff
@@ -114,18 +117,31 @@ protected void beforeIndexDeletion() throws Exception {
         ensureStableCluster(numberOfNodes);
 
         // TODO: this is a temporary solution so that nodes will not base their reaction to a partition based on previous successful results
-        ZenPing zenPing = ((TestZenDiscovery) internalCluster().getInstance(Discovery.class)).getZenPing();
-        if (zenPing instanceof UnicastZenPing) {
-            ((UnicastZenPing) zenPing).clearTemporalResponses();
-        }
+        clearTemporalResponses();
         return nodes;
     }
 
+    protected void clearTemporalResponses() {
```

Comment:
This thing is weird: it clears out the received pings from a single randomly-chosen node in the cluster. It was weird before this change too. But it will go away soon.

Reply:
I agree it's weird. I did not think it was worth investigating further there because this is going away soon.